File: 2022-2023/sourcecode/lab/lab13/OfflineCachingBasicsJava/app/src/main/java/pl/udu/uwr/pum/offlinecachingbasicsjava/data/converters/CreditCardConverter.java
package pl.udu.uwr.pum.offlinecachingbasicsjava.data.converters;

import androidx.room.TypeConverter;

import pl.udu.uwr.pum.offlinecachingbasicsjava.data.response.CreditCard;

public class CreditCardConverter {

    @TypeConverter
    public static String fromCreditCard(CreditCard creditCard){
        return creditCard.getCc_number();
    }

    @TypeConverter
    public static CreditCard toCreditCard(String creditCard){
        return new CreditCard(creditCard);
    }
}

File: 2022-2023/sourcecode/lab/lab13/OfflineCachingBasicsKotlin/app/src/main/java/pl/udu/uwr/pum/offlinecachingbasicskotlin/data/converters/CreditCardConverter.kt
package pl.udu.uwr.pum.offlinecachingbasicskotlin.data.converters

import androidx.room.TypeConverter
import pl.udu.uwr.pum.offlinecachingbasicskotlin.data.reponse.Address
import pl.udu.uwr.pum.offlinecachingbasicskotlin.data.reponse.CreditCard

object CreditCardConverter {
    @TypeConverter
    @JvmStatic
    fun fromCreditCard (creditCard: CreditCard): String{
        return creditCard.cc_number
    }

    @TypeConverter
    @JvmStatic
    fun toCreditCard(creditCard: String): CreditCard {

        return CreditCard(creditCard)
    }
}